osu.Game/Overlays/Settings/Sections/General/UpdateSettings.cs

```diff
@@ -11,24 +11,22 @@ namespace osu.Game.Overlays.Settings.Sections.General
 {
     public class UpdateSettings : SettingsSubsection
     {
-        [Resolved(CanBeNull = true)]
-        private OsuGameBase game { get; set; }
-
         [Resolved(CanBeNull = true)]
         private UpdateManager updateManager { get; set; }
 
         protected override string Header => "Updates";
 
-        [BackgroundDependencyLoader(true)]
-        private void load(Storage storage, OsuConfigManager config)
+        [BackgroundDependencyLoader]
+        private void load(Storage storage, OsuConfigManager config, OsuGameBase game)
         {
             Add(new SettingsEnumDropdown<ReleaseStream>
             {
                 LabelText = "Release stream",
                 Bindable = config.GetBindable<ReleaseStream>(OsuSetting.ReleaseStream),
             });
 
-            if (game != null && updateManager != null)
+            // We shouldn't display the button for the base UpdateManager (without updating logic)
+            if (updateManager != null && updateManager.GetType() != typeof(UpdateManager))
```

Comment:
Not sure I like this relying on types this much. I'd probably add a `public virtual bool CanPerformUpdate { get; }` on the update manager (`=> false` in the base, `=> true` in the derived).

Reply:
I believe doing a virtual property is more flexible, so I'm going with that.
Also the default implementation of `CreateUpdateManager()` in `osu.Game.OsuGame`
 passes no arguments to the constructor of `UpdateManager`, and it'd be better to keep it at that.